server/datastore/mysql/hosts.go

```diff
@@ -4419,6 +4440,11 @@ func (ds *Datastore) OSVersion(ctx context.Context, osVersionID uint, teamID *ui
 		return nil, nil, ctxerr.Wrap(ctx, notFound("OSVersion"))
 	}
 
+	if teamOSVersionNotFound {
+		// If the team OS version was not found, but it exists, we need to return empty state.
+		return nil, &updatedAt, nil
```

Comment:
I believe we still want to return an `OSVersion` here but with a HostCount of 0

Reply:
I think this is cleaner. Returning a full OSVersion with empty fields makes it look like we have a bug.

Reply:
@jacobshandling do you need OS details in the response?  This is when querying OS details for a team that doesn't contain that OS.

Reply:
All I need is the `hosts_count`

Reply:
Ok. Updated here: https://github.com/fleetdm/fleet/pull/16876/files#diff-8103c94796fb91784bbc8c75c1fd026e47eee9db34bd5daf30b0aefcae460558R1908-R1910